Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift

```diff
@@ -178,26 +178,22 @@ open class ChatMessageLayoutOptionsResolver {
         return delay > maxTimeIntervalBetweenMessagesInGroup
     }
     
-    func canShowDeliveryStatus(for message: ChatMessage, in channel: ChatChannel) -> Bool {
-        guard message.isSentByCurrentUser else {
-            // Delivery status can only be shown for a message sent by the current user.
-            return false
-        }
-        
-        guard message.type != .deleted, message.type != .ephemeral else {
-            // Delivery status should not be shown for deleted & ephemeral messages.
-            return false
-        }
+    func canShowDeliveryStatus(
+        for message: ChatMessage,
+        in channel: ChatChannel,
+        allowWhenFailed: Bool
+    ) -> Bool {
```

Comment:
Shouldn't this be open?
```suggestion
    open func canShowDeliveryStatus(
        for message: ChatMessage,
        in channel: ChatChannel,
        allowWhenFailed: Bool
    ) -> Bool {
```

if we have it open, we don't need the `allowWhenFailed` flag. If a customer wants a different behaviour, he can do it by overriding this function and changing the failed state behaviour

Reply:
It's not for customers, it's for us, actually. On the channel list, we can show an error as a delivery status (`allowWhenFailed  == true`) but not on the message list (`allowWhenFailed  == false`). But yeah, let's open it 👍